cli/src/util.rs

```diff
@@ -55,6 +55,9 @@ impl Drop for LogSession {
             {
                 Command::new("open").arg(&self.0).output().unwrap();
             }
+
+            #[cfg(any(debug_assertions, test))]
+            validate_graph_log(&self);
```

Comment:
Very cool that you're verifying the graph logs!

My only suggestion is that I think we should perform this check in an explicit unit test of this functionality. Right now, this `LogSession` struct isn't used during a normal test run; it's just a tool that we use to debug failing tests (via `script/test THE_TEST_NAME -D`). And it's kind of an optional feature of the test suite: you can only use it if you have Graphviz installed on your system.

What do you think about adding a specific unit test in `parser_test.rs`, something like this:

```rust
#[test]
#[cfg(unix)]
fn test_parsing_with_debug_graph_enabled() {
    let mut parser = Parser::new();
    let mut debug_graph_file = tempfile::tempfile().unwrap();
    parser.print_dot_graphs(&debug_graph_file);

    // validate the debug graph 
}	
```

I think that it might be better to perform this validation

Reply:
I like this approach much better. Pushed an update and waiting for CI feedback.